x-pack/platform/plugins/shared/alerting/server/backfill_client/backfill_client.ts

```diff
@@ -331,6 +364,50 @@ export class BackfillClient {
       );
     }
   }
+
+  public async findOverlappingBackfills({
+    ruleId,
+    start,
+    end,
+    savedObjectsRepository,
+    actionsClient,
+  }: {
+    ruleId: string;
+    start: Date;
+    end: Date;
+    savedObjectsRepository: ISavedObjectsRepository;
+    actionsClient: ActionsClient;
+  }) {
+    const adHocRuns: Array<SavedObjectsFindResult<AdHocRunSO>> = [];
+
+    // Create a point in time finder for efficient pagination
+    const adHocRunFinder = await savedObjectsRepository.createPointInTimeFinder<AdHocRunSO>({
+      type: AD_HOC_RUN_SAVED_OBJECT_TYPE,
+      perPage: 100,
+      hasReference: [{ id: ruleId, type: RULE_SAVED_OBJECT_TYPE }],
+      filter: `
+        ad_hoc_run_params.attributes.start <= "${end.toISOString()}" and
+        ad_hoc_run_params.attributes.end >= "${start.toISOString()}"
+      `,
+    });
+
+    try {
+      // Collect all results using async iterator
+      for await (const response of adHocRunFinder.find()) {
+        adHocRuns.push(...response.saved_objects);
+      }
+    } finally {
+      // Make sure we always close the finder
+      await adHocRunFinder.close();
+    }
+
+    return adHocRuns.map((data) =>
```

Comment:
It doesn't look like we use any of the information returned except for the schedule, is that correct? Can we change this function to be `findOverlappingBackfillSchedules` and just return the schedule information? The backfill object is fairly large and if we can limit the information we're passing around, we should.